src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/JsonReaderExtensions.cs

```diff
@@ -50,7 +49,7 @@ public static TArg ReadProperties<TArg>(this JsonReader reader, Func<string, TAr
             return arg;
         }
 
-        public static bool TryReadNextProperty<TReturn>(this JsonReader reader, string propertyName, out TReturn value)
+        public static bool TryReadNextProperty<TReturn>(this JsonReader reader, string propertyName, [NotNullWhen(true)] out TReturn? value)
```

Comment:
❗ This contract is not correct (JSON property values can be null, and the cast to `TReturn` won't fail for this). Unless additional validation is added, the method should be:

```suggestion
        public static bool TryReadNextProperty<TReturn>(this JsonReader reader, string propertyName, out TReturn? value)
```